libarchive/archive_read_support_format_rar5.c

```diff
@@ -1715,9 +1715,28 @@ static int process_head_file(struct archive_read* a, struct rar5* rar,
 	/* If we're currently switching volumes, ignore the new definition of
 	 * window_size. */
 	if(rar->cstate.switch_multivolume == 0) {
-		/* Values up to 64M should fit into ssize_t on every
-		 * architecture. */
-		rar->cstate.window_size = (ssize_t) window_size;
+		if(rar->cstate.window_size < (ssize_t) window_size) {
```

Comment:
window_size should still be updated, even if no reallocation happened? I think this whole block would become more readable if it was:
```
if (switch_multivolume == 0) {
    if (rar->cstate.window_size < (ssize_t)window_size && rar->cstate.window_buf) {
        / *realloc logic */
    }
    rar->cstate.window_size = (ssize_t)window_size;
}
```